kmk/modules/encoder.py

```diff
@@ -122,6 +122,7 @@ def __init__(self, pin_a, pin_b, pin_button=None, is_inverted=False, divisor=Non
         self.pin_a = EncoderPin(pin_a)
         self.pin_b = EncoderPin(pin_b)
         self.pin_button = (
+            EncoderPin(*pin_button) if isinstance(pin_button, tuple) else
```

Comment:
Make that a proper named argument instead of overloading `pin_button` with a tuple.

Reply:
This is not possible with how you have defined the pins config initially.  My concern is that there are users who have all kinds of config specified in their code.py as anonymous tuples of fixed length.  If I introduce an extra named argument, while this code would work - it would break your users configs in code.py.

Here is my config from code.py before any change to my PR:
`encoder_handler.pins = ((board.VOLC_A, board.VOLC_B, (board.VOLC_PUSH, True, digitalio.Pull.DOWN), False, 2),)
`
And here is my config after I introduce this proper named argument:
`encoder_handler.pins = ((board.VOLC_A, board.VOLC_B, None, (board.VOLC_PUSH, True, digitalio.Pull.DOWN), False, 2),)`

See how I have extended the length of this pins config tuple by one extra argument which is "None" for the original "pin_button" argument?
This will break users configs who have the regular tuple.  Because your code expects a tuple in this way and passes this directly into the constructor of GPIOEncoder like this:
`new_encoder = GPIOEncoder(*pins)
`

This is why I went with overloading "pin_button" with a tuple.
Is there anything else that may be done to fix this?  I don't want to break compatibility with user configs.  And my original PR does not break compatibility.

Reply:
Here's a mockup of what I had in mind:
```python
class GPIOEncoder(BaseEncoder):
    def __init__(self, pin_a, pin_b, pin_button=None, is_inverted=False, divisor=None, button_pull=digitalio.Pull.UP):
        # ...
        if pin_button:
            self.pin_button = EncoderPin(pin_button, button_type=True, button_pull=button_pull)
        else:
            self.pin_button = None
```

Can be used like this:
```
encoder_handler.pins = (
    (board.GP0, board.GP1, board.GP2, False, 2, digitalio.Pull.DOWN),
    (board.GP3, board.GP4, board.GP5),
)
```

Doesn't break backwards compatibility, although we have to break it eventually to get rid of some of the old garbage code -- like guessing a constructor from the contents of a tuple, instead of using the proper constructor in the first place.

Reply:
Implemented, tested and documented the change